optimum/onnxruntime/base.py

```diff
@@ -552,4 +560,19 @@ def filter_out_output(output_name):
                 if use_torch:
                     loss = torch.from_numpy(loss).to(self.device)
 
-        return Seq2SeqLMOutput(loss=loss, logits=logits, past_key_values=past_key_values)
+        # TODO: this is extremely ugly and unreadable. What if cross-attention k/v change?
+        # Tuple of tuple of length `n_layers`, with each tuple of length equal to:
+        # * 4 for the decoder without cache (k/v of self-attention + k/v of cross-attention)
+        # * 2 for the decoder with cache (k/v of self-attention as cross-attention cache is constant)
+        if self.use_past is False:
```

Comment:
Not easily readable imo

Reply:
Agreed, this can be improved later. Let me know if the comment can be improved.

Currently some pkv inputs and outputs are exactly the same, so not useful and they add an unnecessary overhead.